pkg/controller/build/build_controller.go

```diff
@@ -792,139 +811,115 @@ func (ctrl *Controller) postBuildCleanup(pool *mcfgv1.MachineConfigPool, ignoreM
 }
 
 // Marks a given MachineConfigPool as build successful and cleans up after itself.
-func (ctrl *Controller) markBuildSucceeded(ps *poolState) error {
-	klog.Infof("Build succeeded for MachineConfigPool %s, config %s", ps.Name(), ps.CurrentMachineConfig())
-
-	pool := ps.MachineConfigPool()
-
-	// Get the final image pullspec.
-	imagePullspec, err := ctrl.imageBuilder.FinalPullspec(pool)
-	if err != nil {
-		return fmt.Errorf("could not get final image pullspec for pool %s: %w", ps.Name(), err)
-	}
-
-	if imagePullspec == "" {
-		return fmt.Errorf("image pullspec empty for pool %s", ps.Name())
-	}
-
-	// Perform the post-build cleanup.
-	if err := ctrl.postBuildCleanup(pool, false); err != nil {
-		return fmt.Errorf("could not do post-build cleanup: %w", err)
-	}
-
+func (ctrl *Controller) markBuildSucceeded(mosb *mcfgv1alpha1.MachineOSBuild) error {
 	// Perform the MachineConfigPool update.
 	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		mcp, err := ctrl.mcfgclient.MachineconfigurationV1().MachineConfigPools().Get(context.TODO(), ps.Name(), metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		ps := newPoolState(mcp)
 
-		// Set the annotation or field to point to the newly-built container image.
-		klog.V(4).Infof("Setting new image pullspec for %s to %s", ps.Name(), imagePullspec)
-		ps.SetImagePullspec(imagePullspec)
+		// need to do the below with the mosb
+		/*
+			ps := newPoolState(mcp)
 
-		// Remove the build object reference from the MachineConfigPool since we're
-		// not using it anymore.
-		ps.DeleteBuildRefForCurrentMachineConfig()
+			// Set the annotation or field to point to the newly-built container image.
+			klog.V(4).Infof("Setting new image pullspec for %s to %s", ps.Name(), imagePullspec)
+			ps.SetImagePullspec(imagePullspec)
```

Comment:
note: This value should be provided by the FinalPullspec() method on both the ImageBuilder and CustomPodBuilder objects. The reason is because both of those objects return the digested image pullspec instead of the tagged image pullspec.